src/components/LmsConfigurations/index.test.jsx

```diff
@@ -83,21 +113,48 @@ describe('<LmsConfigurations /> ', () => {
     expect(wrapper.find('BlackboardIntegrationConfigForm').props().config).toEqual(wrapper.state().blackboardConfig);
   });
 
+  it('get the Canvas LMS configuration when present', async () => {
+    LmsApiService.fetchMoodleConfig.mockRejectedValue(notFoundMoodleResponse);
+    LmsApiService.fetchBlackboardConfig.mockRejectedValue(notFoundBlackboardResponse);
+    LmsApiService.fetchCanvasConfig.mockResolvedValue(canvasResponse);
+    LmsApiService.fetchSuccessFactorsConfig.mockRejectedValue(notFoundSuccessFactorsResponse);
+
+    const wrapper = mount(<LmsConfigurations enterpriseId="testEnterpriseId" />);
+    await waitForAsync();
+    wrapper.update();
+    expect(wrapper.state().canvasConfig).toEqual(canvasResponse.data.results[0]);
+    expect(wrapper.find('CanvasIntegrationConfigForm').props().config).toEqual(wrapper.state().canvasConfig);
```

Comment:
I'd also love to add a test to show that the config data is actually being displayed on the page rather than just in the state. 
I would do something like:
`expect(wrapper.find('CanvasIntegrationConfigForm').text()).toContain(<some string from your mock response>)`

Reply:
I feel like that's already covered by the component tests. Ie if you look at the CanvasIntegrationConfigForm tests they will comprehensively cover the fields therein and making sure they auto-populate based on props/state.

The goal of this test is just to make sure props are being passed. Like the other test above using `CollapsibleAdvanced`, the differentiator in the logic here isn't exclusively/necessarily around which forms we display, but also what data we pass through to it.

Reply:
We may want a full end-to-end test somewhere, but I don't feel this test suite is the place to put it.
Logic-wise if we verify the props are getting sent from the index and we thoroughly test the form data/passed prop logic in the components it uses then we have 100% coverage. There is certainly an argument that can be made about an end-to-end test that tests that flow truly and completely working, but similar to your overall PR comment, I'd say that's a topic for debate and addressing at a later time.

Reply:
I don't consider testing the UI to be an end-to-end test (though I agree that is also useful). In general, the React community is moving **strongly** away from testing what's in state or props and instead testing what's in the UI, and I do think we should be testing what the user sees. So, yes, it's an integration test rather than a unit test, but I think the test should exist. I'm OK with us opening a ticket for it.

Reply:
ah I completely forgot that type of test ("integration"). We can open a ticket as I think we should add these for all sidebar items that don't already have them. If you happen to have - and don't feel obligated to go searching - some links to standards being discussed or the community talking about the testing strategies here I'd love to take a look. I've used snapshots as well as trying to dig into the DOM to do actual integration tests and I've found them really flaky to the point of being useless, however React moves fast, so there may be much better solutions now.